openfisca_uk_data/datasets/frs/frs.py

```diff
@@ -1,87 +1,635 @@
+            benefits.BENAMT[benefits.BENEFIT == code]
+            .groupby(benefits.person_id)
+            .sum()
+            .reindex(index=person.index)
+            .fillna(0)
+            .values
+            * 52
+        )
+
+    frs["BSP_reported"] = (
+        benefits.BENAMT[benefits.BENEFIT.isin((6, 9))]
+        .groupby(benefits.person_id)
+        .sum()
+        .reindex(index=person.index)
+        .fillna(0)
+        .values
+        * 52
+    )
+
+    frs["winter_fuel_allowance_reported"][...] = (
+        np.array(frs["winter_fuel_allowance_reported"]) / 52
+    )
+
+    frs["SSP_reported"] = person.SSPADJ * 52
+
+    frs["student_loans"] = person.TUBORR
+
+    frs["student_payments"] = person[["ADEMAAMT", "CHEMAAMT", "ACCSSAMT"]].sum(
+        axis=1
+    ) * 52 + person[["GRTDIR1", "GRTDIR2"]].sum(axis=1)
+
+    frs["council_tax_benefit_reported"] = (
+        (person.HRPID == 1)
+        * pd.Series(
+            household.CTREBAMT[person.household_id].values, index=person.index
+        ).fillna(0)
+        * 52
+    )
+
+
+def add_expenses(
+    frs: h5py.File,
+    person: DataFrame,
+    household: DataFrame,
+    maintenance: DataFrame,
+    mortgage: DataFrame,
+    childcare: DataFrame,
+    pen_prov: DataFrame,
+):
+    """Adds expense variables
+
+    Args:
+        frs (h5py.File)
+        person (DataFrame)
+        household (DataFrame)
+        maintenance (DataFrame)
+        mortgage (DataFrame)
+        childcare (DataFrame)
+        pen_prov (DataFrame)
+    """
+    frs["maintenance_expenses"] = (
+        pd.Series(
+            np.where(
+                maintenance.MRUS == 2, maintenance.MRUAMT, maintenance.MRAMT
+            )
+        )
+        .groupby(maintenance.person_id)
+        .sum()
+        .reindex(person.index)
+        .fillna(0)
+        * 52
+    )
+
+    frs["housing_costs"] = (
+        np.where(
+            household.GVTREGNO != 13, household.GBHSCOST, household.NIHSCOST
+        )
+        * 52
+    )
+    frs["rent"] = household.HHRENT * 52
+    frs["mortgage_interest_repayment"] = household.MORTINT * 52
+    mortgage_capital = np.where(
+        mortgage.RMORT == 1, mortgage.RMAMT, mortgage.BORRAMT
+    )
+    mortgage_capital_repayment = (
+        (mortgage_capital / mortgage.MORTEND)
+        .groupby(mortgage.household_id)
+        .sum()
+        .reindex(household.index)
+        .fillna(0)
+    )
+    frs["mortgage_capital_repayment"] = mortgage_capital_repayment
+
+    frs["childcare_expenses"] = (
+        childcare.CHAMT[childcare.COST == 1][childcare.REGISTRD == 1]
+        .groupby(childcare.person_id)
+        .sum()
+        .reindex(person.index)
+        .fillna(0)
+    ) * 52
+
+    frs["private_pension_contributions"] = (
+        pen_prov.PENAMT[pen_prov.STEMPPEN.isin((5, 6))]
+        .groupby(pen_prov.person_id)
+        .sum()
+        .reindex(person.index)
+        .fillna(0)
+        .clip(0, pen_prov.PENAMT.quantile(0.95))
```

Comment:
is this what ukmod does?

Reply:
I think so, from the input microdata documentation:
```
gen xpp=penamt*52/12 if stemppen==5|stemppen==6 
 su xpp if xpp>0, d
 replace xpp = r(p95) if xpp > r(p95)
```

Reply:
Do they say why they clip that variable specifically? It'd be good to document this choice somewhere.